src/jit/codegencommon.cpp

```diff
@@ -10706,7 +10688,39 @@ void CodeGen::genRestoreCalleeSavedFltRegs(unsigned lclFrameSize)
             offset -= XMM_REGSIZE_BYTES;
         }
     }
+    genVzeroupperIfNeeded();
+}
+
+// Generate Vzeroupper instruction as needed to zero out upper 128b-bit of all YMM registers so that the
+// AVX/Legacy SSE transition penalties can be avoided
+//
+// Params
+//   check256bitOnly  - true to check if the function contains 256-bit AVX instruction and generate Vzeroupper
+//      instruction, false to check if the function contains AVX instruciton (either 128-bit or 256-bit).
+//
+void CodeGen::genVzeroupperIfNeeded(bool check256bitOnly /* = true*/)
+{
+#ifdef FEATURE_AVX_SUPPORT
+    if (compiler->getSIMDInstructionSet() == InstructionSet_AVX)
+    {
```

Comment:
Here too we don't need to check for instruction set because lower is making that check before setting flags on emitter.  Instead we should assert.

bool emitVzeroUpper;
if (check256bitOnly)
{
        emitVzeroUpper  = getEmitter()->Contains256bitAVX();
}
else
{       
       emitVzeroUpper = getEmitter()->ContainsAVX();
}

if (emitVzeroUpper)
{
         assert(compiler->getSIMDInstructionSet() == InstructionSet_AVX);                
         instGen(INS_vzeroupper);
}